variables.tf

```
@@ -6,6 +6,11 @@ variable "additional_ecr_policy_statements" {
       identifiers = list(string)
     })
     actions = list(string)
+    condition = optional(object({
+      test     = optional(string)
+      variable = optional(string)
+      values   = optional(list(string))
```

```suggestion
      test     = string
      variable = string
      values   = list(string)
```

The `condition` block can be optional but if used then it needs all three fields to be populated.

---

quite right, this also made me realise that it did not allow for a list of multiple conditions as it was now. Updated the function to allow a list of conditions (optionally) and made the object variables required